### PR TITLE
feat: remove liquidity single side

### PR DESCRIPTION
**Pull request by dongnguyen9186**

### CODE REVIEW

**Review by 00xSam — APPROVED** [review]
Help update version number and update changelog 🙏🏿

File: ts-client/src/dlmm/index.ts [hunk 1]
```diff
@@ -4545,6 +4654,141 @@ export class DLMM {
...
+      this.lbPair.parameters,
+      this.lbPair.vParameters,
+      maxAmountIn
+    );
+
+    maxAmountIn = maxAmountIn.add(maxFee);
+
+    if (amount0.gt(maxAmountIn)) {
+      return {
+        withdrawAmount: maxAmountOut,
```
Comment (andrewsource147) [inline ts-client/src/dlmm/index.ts:4703]: = maxAmountOut + amount1

File: ts-client/src/dlmm/index.ts [hunk 2]
```diff
@@ -4545,6 +4654,141 @@ export class DLMM {
...
+      this.lbPair.vParameters,
+      amount0
+    );
+    const amount0AfterFee = amount0.sub(fee);
+    const amountOut = isWithdrawForY
+      ? mulShr(price, amount0AfterFee, SCALE_OFFSET, Rounding.Down)
+      : shlDiv(amount0AfterFee, price, SCALE_OFFSET, Rounding.Down);
+
+    return {
+      withdrawAmount: amount1.add(amountOut.gt(maxAmountOut) ? maxAmountOut : amountOut),
```
Comment (andrewsource147) [inline ts-client/src/dlmm/index.ts:4718]: = amount1 + swap(amount0 -> amountOut)

File: ts-client/src/dlmm/index.ts [hunk 3]
```diff
@@ -4545,6 +4654,141 @@ export class DLMM {
...
+      if (shareToRemove.eq(new BN(0))) {
+        continue;
+      }
+
+      const binId = new BN(position.lowerBinId).add(new BN(idx));
+      const binArrayIndex = binIdToBinArrayIndex(binId);
+      const binArray = binArrayIndex.eq(lowerBinArrayIdx)
+      ? lowerBinArray
+      : upperBinArray;
+      const bin = getBinFromBinArray(binId.toNumber(), binArray);
```
Comment (andrewsource147) [inline ts-client/src/dlmm/index.ts:4767]: throw error here if binArray is undefined or null

**Pull request merged**